### PR TITLE
add backup manager to scheduled automatic backups

### PR DESCRIPTION
**Pull request by patricioxavier8**
This PR introduces improvements to the ScheduledBackupManager to manage the scheduling of automatic backups, ensuring persistence even when the app is closed and reopened.

Key Changes:

The last backup execution time is saved in UserDefaults with the key lastBackupTime.

Timer.scheduledTimer(withTimeInterval:repeats:block:) is used to set up a one-time timer that fires after timeUntilNextBackup seconds, calculated from the last backup.
The timer's block executes performBackup(), then sets up the recurring timer.

After the first backup, a recurring timer is set up using Timer.publish(every:on:in:) to run at regular intervals based on the user's selected frequency.

resumeBackupScheduler() is implemented to retrieve the backup frequency from UserDefaults and restart the timer when the app resumes.

### CODE REVIEW

File: Shared/Services/Backups/ScheduledBackupManager.swift [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    
+    private func performBackup() {
+        Task {
+            do {
+                await backupsService.loadAllDevices()
+                backupsService.loadFoldersToBackup()
+                try await backupsService.startBackup { progress in}
+                userDefaults.set(Date(), forKey: LAST_BACKUP_TIME_KEY)
+            } catch {
+
```
Comment (PixoDev) [inline Shared/Services/Backups/ScheduledBackupManager.swift:58]: Do something with the possible error here, something in the UI should be displayed at least an NSAlert

File: Shared/Services/Backups/ScheduledBackupManager.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        
+    }
+    
+    
+    private func performBackup() {
+        Task {
+            do {
+                await backupsService.loadAllDevices()
+                backupsService.loadFoldersToBackup()
+                try await backupsService.startBackup { progress in}
```
Comment (PixoDev) [inline Shared/Services/Backups/ScheduledBackupManager.swift:55]: Does this mean that when an scheduled backup happens, I don't see the progress?

File: InternxtDesktop/AppDelegate.swift [hunk 3]
```diff
@@ -283,6 +283,7 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         } else {
             self.openWidget(delayed: true)
         }
+        ScheduledBackupManager.shared.resumeBackupScheduler()
```
Comment (PixoDev) [inline InternxtDesktop/AppDelegate.swift:286]: Can we recheck the backup scheduler every time the user opens the widget also? just to be sure

**Review by PixoDev — CHANGES_REQUESTED** [review]

File: Shared/Services/Backups/ScheduledBackupManager.swift (hunk 2 above)
Comment (patricioxavier8) [reply Shared/Services/Backups/ScheduledBackupManager.swift:55]: Fixed, progress is now showing.

File: Shared/Services/Backups/ScheduledBackupManager.swift (hunk 1 above)
Comment (patricioxavier8) [reply Shared/Services/Backups/ScheduledBackupManager.swift:58]: Fixed

**Comment by PixoDev** [comment]
Great job here, scheduled backups works as expected, it triggers a backup at the specified time.

I found a corner case where the time displayed is not correct tho

1. I have scheduled backups each hour, next one is at 10:13 let's say
2. My pc goes to sleep
3. I open my pc again, at 15:43
4. The displayed next backup, is going to be 10:13, not 16:13

If I open the widget, the date is updated tho, but If I had the window opened, then it will display previous not updated data

Maybe this can help us to tackle that corner case:

https://developer.apple.com/documentation/swiftui/scenephase

**Review by PixoDev — APPROVED** [review]

**Pull request merged**